integration/yarn/src/main/java/alluxio/yarn/ContainerAllocator.java

```diff
@@ -153,7 +153,7 @@ private void requestContainers(int numContainersToRequest) throws Exception {
       priority = Priority.newInstance(101);
     }
 
-    if (hosts.length * mMaxContainersPerHost < numContainersToRequest) {
+    if (hosts.length * mMaxContainersPerHost < mTargetNumContainers - mAllocatedContainers.size()) {
```

Comment:
yea, it's basically just a [stale version](https://github.com/Alluxio/alluxio/pull/4643/files#diff-2038a4077b21a39bb42af04d3de1df2bR108) of `mTargetNumContainers - mAllocatedContainers.size()`